release_notes/ocp_3_9_release_notes.adoc

```diff
@@ -0,0 +1,434 @@
+xref:../install_config/install/planning.adoc#install-config-install-planning[Installing
+a Cluster] topics in the
+xref:../install_config/index.adoc#install-config-index[Installation and
+Configuration] documentation.
+
+To upgrade to this release from a previous version, see the
+xref:../install_config/upgrading/index.adoc#install-config-upgrading-index[Upgrading
+a Cluster] topics in the
+xref:../install_config/index.adoc#install-config-index[Installation and
+Configuration] documentation.
+
+[[ocp-39-new-features-and-enhancements]]
+== New Features and Enhancements
+
+This release adds improvements related to the following components and concepts.
+
+[[ocp-39-container-orchestration]]
+=== Container Orchestration
+
+[[ocp-39-crio]]
+==== CRI-O
+
+CRI-O v1.0 is a lightweight, native Kubernetes container runtime interface. By
+design, it provides only the runtime capabilities needed by the kubelet. CRI-O is
+designed to be part of Kubernetes and evolve in lock-step with the platform.
+
+CRI-O brings:
+
+* A minimal and secure architecture.
+* Excellent scale and performance.
+* The ability to run any Open Container Initiative (OCI) or docker image.
+* Familiar operational tooling and commands.
+
+image::crio-3-7.png[CRI-O]
+
+To install and run CRI-O alongside `docker`, set the following in the
+`[OSEv3:vars]` section
+xref:../install_config/install/advanced_install.adoc#configuring-ansible[Ansible inventory file] during cluster installation:
+
+----
+openshift_use_crio=true
+----
+
+This setting pulls the *openshift3/cri-o* system container image from the
+link:https://access.redhat.com/containers[Red Hat Registry] by default. If you
+want to use an alternative CRI-O system container image from another registry,
+you can also override the default using the following variable:
+
+----
+openshift_crio_systemcontainer_image_override=<registry>/<repo>/<image>:<tag>
+----
+
+[NOTE]
+====
+The `atomic-openshift-node` service must be RPM- or system container-based when
+using CRI-O; it cannot be `docker` container-based. The installer protects again
+using CRI-O with `docker` container nodes and will halt installation if
+detected.
+====
+
+When CRI-O use is enabled, it is installed alongside `docker`, which currently
+is required to perform build and push operations to the reigstry. Over time,
+temporary `docker` builds can accumulate on nodes. You can optionally set the
+following to enable garbage collection, which adds a daemonset to clean out the
+builds:
+
+----
+openshift_crio_enable_docker_gc=true
+----
+
+When enabled, it will run garbage collection on all nodes by default. You can
+also limit the running of the daemonset on specific nodes by setting the
+following:
+
+----
+openshift_crio_docker_gc_node_selector={'runtime': 'cri-o'}
+----
+
+For example, the above would ensure it is only run on nodes with the `runtime:
+cri-o` label. This can be helpful if you are running CRI-O only on some nodes,
+and others are only running `docker`.
```

Comment:
Do we have an example setup or recommendations on how to migrate to CRI-O? 
Should customers migrate 10% or 50% of the infrastructure on day 1? What/how does the upgrade handle this? 

@sdodson

Reply:
The upgrade doesn't migrate to cri-o to my knowledge. The intent is that cri-o nodes are net new as far as I know. @ashcrow

Reply:
Yup, they are net new. One could bring up X nodes that are cri-o and put them in the cluster then slowly (or quickly) replace traditional runtimes nodes with cri-o nodes.